### ISSUE TITLE
String parameters to functions: should fail when type is known to be any

### ISSUE THREAD

**Issue by HamletDRC** [Microsoft/tslint-microsoft-contrib#193]
This should fail... because we _know_ that the type is 'any'. It is not 'any' because we somehow failed to determine its type. 

```
export module actions {
    class AsyncAction {

        private delegate;
        constructor(delegate) {
            this.delegate = delegate;
        }

        public invoke(): number {
            return setTimeout(this.delegate, 0);
        }
    }
}
```

**Comment by JoshuaKGoldberg** [comment]
@HamletDRC I believe `--noImplicitAny` should catch this. If it doesn't, I think this should be filed in TSLint core.
